.github/workflows/lint-autoformat.yml

```diff
@@ -0,0 +1,46 @@
+name: Apply lint suggestions
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+    branches: [main]
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches: [gh/**/base]
+
+jobs:
+  lintrunner-autoformat:
+    permissions:
+      contents: read
+      pull-requests: write
+    runs-on: lf.linux.2xlarge
+    if: ${{ github.repository_owner == 'pytorch' }}
+    steps:
+      - name: Checkout PyTorch
+        uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
+        with:
+          submodules: true
+          fetch-depth: 0
+      - name: Setup miniconda
+        uses: pytorch/test-infra/.github/actions/setup-miniconda@main
+        with:
+          python-version: "3.10"
+      - name: Run lintrunner (nonretryable)
+        continue-on-error: true
+        run: |
+          export ADDITIONAL_LINTRUNNER_ARGS="format"
+          .github/scripts/lintrunner.sh
```

Comment:
Please verify if lintrunner.sh is coming from the main branch, not PR, otherwise it's security issue with pull_request_target.

Reply:
Yeah, this is a valid concern.  As the job needs to checkout the PR to run lint, I'm thinking we either need to 

1. Checkout main PyTorch another time, store it in a different directory, then copy `lintrunner.sh` from there or
2. Just download the script from upstream https://raw.githubusercontent.com/pytorch/pytorch/refs/heads/main/.github/scripts/lintrunner.sh

Do you know if there is a better alternative to get the script?

Reply:
So does lintrunner.sh come from the PR for pull_request_target?

Right now I'm confused how it actually works, see this comment for another workflow https://github.com/pytorch/pytorch/pull/133861#issuecomment-2373030151

Reply:
I think I figured out why https://github.com/pytorch/pytorch/pull/133861#issuecomment-2373030151 works - I updated that comment.

This PR I don't think will work, it will always check code form main for pull_request_target, not the actual PR code.

Reply:
For the security issue (assuming we resolve the problem with always checking main) copying `lintrunner.sh` is not enough, as it runs other PyTorch files, like torch/utils/data/datapipes/gen_pyi.py.

Reply:
How to make it work - probably the easiest thing is to create a reusable action in test-infra, that will use `github.event.pull_request.head.sha`, and call it from a PyTorch workflow: similar to how https://github.com/pytorch/pytorch/pull/133861/files calls ethanis/nitpicker@v1

Reply:
Oh, TIL, I completely misread what `pull_request_target` checkout.  It's indeed checking out the PR base commit (main).  The doc says this clearly https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target and I should have pay more attention there.  So, the solution you bring up makes sense.  

Because of the difficulty of manually bringing other files like `torch/utils/data/datapipes/gen_pyi.py`, I think we could do something like this in the workflow:

1. Use `pull_request_target`.
2. Have a job to checkout `github.event.pull_request.head.sha`, make sure to **downgrade its permission to read-only** like a regular PR.  We can safely run `lintrunner.sh` from the PR here.
3. Get the output of git diff, upload it as an artifacts.
4. Have another job with full read/write permission and also checkout `github.event.pull_request.head.sha`, then download the diff from (3) and git apply it (I assume this would be safe).  We can run `parkerbxyz/suggest-changes` here.

Reply:
An important detail is how https://github.com/pytorch/pytorch/blob/main/.github/actions/checkout-pytorch/action.yml is implemented: it has `github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha`. Without `event_name` check, it could actually use the PR code even for `pull_request_target`.

Reply:
@huydhn Can we just checkout `github.event.pull_request.head.sha` into a different directory and run code from main on it?